### ISSUE TITLE
`parcel build` with web worker produces code that cannot execute

### ISSUE THREAD

**Issue by jeffpeck10x** [parcel-bundler/parcel#6269]
# 🐛 bug report

`parcel build` with web worker produces code that cannot execute

```
  const worker = new Worker(
      new URL("./workers/my-worker.ts", import.meta.url),
      { type: "module" }
    );
```

```
$ parcel build ./src/index.ts --no-source-maps --no-optimize --no-cache
```

The result in `dist/module/index.js`:
```
$37fff9a7ec0a7105da2f7eb5f3df596b$exports = $53630ef4c3b85e8cc60065ddf3d18aba$init()("../main/my-worker.ed75ea8e.js");
```

which fails when imported in another application with:
```
Error: Origin not found
```

Why? Because `getBundleURL()` is returning a `/`:
https://github.com/parcel-bundler/parcel/blob/v2/packages/runtimes/js/src/bundle-url.js#L12

And then `getOrigin()` fails to regex match the origin:
https://github.com/parcel-bundler/parcel/blob/v2/packages/runtimes/js/src/bundle-url.js#L39

I am wondering if I am missing something, but I am not sure what to even try next.

Honestly, I would really appreciate if the answer is: "Parcel does not yet support web workers" just so I don't have to spend time trying to make this work, and can focus on a different solution.

## 🎛 Configuration (.babelrc, package.json, cli command)

.babel.config.json
```json
{
  "plugins": [
    "@parcel/babel-plugin-transform-runtime",
    ["inline-svg", { "disableSVGO": true }]
  ],
  "presets": [
    "@parcel/babel-preset-env",
    "@babel/preset-react",
    "@babel/preset-typescript"
  ]
}

```

## 🤔 Expected Behavior

`parcel build` will produce code that will not crash when encountering a web worker.

## 😯 Current Behavior

```
Error: Origin not found
```

## 🔦 Context

Cannot use web workers with parcel2 at the moment.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-beta.2
| Node             | 12.18.3
| npm/Yarn         | 1.22.10
| Operating System | MacOS 10.15.7

**Comment by jeffpeck10x** [comment]
I have found a workaround, prepending "data-url:" to the path:
```
    const worker = new Worker(
      new URL("data-url:./workers/my-worker.ts", import.meta.url),
      { type: "module" }
    );
```

This will put the contents of `my-worker.ts` into a blob. I actually found this in an example in the parcel documentation.

---

There is a caveat though: _This will not work with import statements in the worker._

Even though doing `import workerScript as "data-url:./workers/my-worker.ts"` will bundle the imported scripts, doing the same thing inside a `new Worker` does not. And I have not been able to find a way to pass the full blob to `new Worker`. I am not clear as to why, but if I try to manually make a Blob URL from that import (or any text, oddly), the import statement fails.

tl;dr I cannot find a workaround that will allow import statements.

This is good enough for now, but there does not seem to be a straightforward way to use web workers with parcel2.

**Comment by jeffpeck10x** [comment]
My workaround does not work in Firefox:
```
SyntaxError: export declarations may only appear at top level of a module
```

**Comment by devongovett** [comment]
Please try in the latest nightly. We recently added support for ESM workers.

**Comment by egasimus** [comment]
@devongovett I'm experiencing the same issue as @jeffpeck10x. Workarounds breaking `import` or Firefox are a no-go.

**Comment by egasimus** [comment]
Found a fix that works for me. I hope it's upstream-worthy - please review when you have the time. ✌🏿
